pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TopicsBase.java

```diff
@@ -0,0 +1,599 @@
+            }
+        }
+        return result;
+    }
+
+    private SchemaData getSchemaData(String keySchema, String valueSchema) {
+        try {
+            if ((keySchema == null || keySchema.isEmpty()) && (valueSchema == null || valueSchema.isEmpty())) {
+                return null;
+            } else {
+                SchemaInfo keySchemaInfo = (keySchema == null || keySchema.isEmpty())
+                        ? StringSchema.utf8().getSchemaInfo() : ObjectMapperFactory.getThreadLocal()
+                        .readValue(Base64.getDecoder().decode(keySchema), SchemaInfo.class);
+                SchemaInfo valueSchemaInfo = (valueSchema == null || valueSchema.isEmpty())
+                        ? StringSchema.utf8().getSchemaInfo() : ObjectMapperFactory.getThreadLocal()
+                        .readValue(Base64.getDecoder().decode(valueSchema), SchemaInfo.class);
+                if (null == keySchemaInfo.getName()) {
+                    keySchemaInfo.setName(keySchemaInfo.getType().toString());
+                }
+                if (null == valueSchemaInfo.getName()) {
+                    valueSchemaInfo.setName(valueSchemaInfo.getType().toString());
+                }
+                SchemaInfo schemaInfo = KeyValueSchemaInfo.encodeKeyValueSchemaInfo("KVSchema-"
+                                + topicName.getPartitionedTopicName(),
+                        keySchemaInfo, valueSchemaInfo,
+                        KeyValueEncodingType.SEPARATED);
+                return SchemaData.builder()
+                        .data(schemaInfo.getSchema())
+                        .isDeleted(false)
+                        .user("Rest Producer")
+                        .timestamp(System.currentTimeMillis())
+                        .type(schemaInfo.getType())
+                        .props(schemaInfo.getProperties())
+                        .build();
+            }
+        } catch (IOException e) {
+            if (log.isDebugEnabled()) {
+                log.warn("Fail to parse schema info for rest produce request with key schema {} and value schema {}"
+                        , keySchema, valueSchema);
+            }
+            return null;
+        }
+    }
+
+    // Convert message to ByteBuf
+    public ByteBuf messageToByteBuf(Message message) {
+        checkArgument(message instanceof MessageImpl, "Message must be type of MessageImpl.");
+
+        MessageImpl msg = (MessageImpl) message;
+        MessageMetadata messageMetadata = msg.getMessageBuilder();
+        ByteBuf payload = msg.getDataBuffer();
+        messageMetadata.setCompression(CompressionCodecProvider.convertToWireProtocol(CompressionType.NONE));
+        messageMetadata.setUncompressedSize(payload.readableBytes());
+
+        ByteBuf byteBuf = null;
+        try {
+            byteBuf = Commands.serializeMetadataAndPayload(Commands.ChecksumType.Crc32c, messageMetadata, payload);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Please avoid doing this. Can you add a log statement properly?

Reply:
my bad, updated to proper logging.